### PR TITLE
make near-vm-runner build /w --no-default-features

### PR DESCRIPTION
**Pull request by nagisa**
I wanted to utilize the features here to disable and not deal with
wasmer0 in some experiments, but was surprised to find out the code
simply does not compile if the feature is disabled.

### CODE REVIEW

File: runtime/near-vm-runner/src/cache.rs [hunk 1]
```diff
@@ -111,12 +117,12 @@ impl fmt::Debug for MockCompiledContractCache {
 #[cfg(not(feature = "no_cache"))]
 const CACHE_SIZE: usize = 128;
 
-#[cfg(not(feature = "no_cache"))]
+#[cfg(all(feature = "wasmer0_vm", not(feature = "no_cache")))]
 static WASMER_CACHE: once_cell::sync::Lazy<
```
Comment (matklad) [inline runtime/near-vm-runner/src/cache.rs:121]: Can we move these inside the corresponding modules?

File: runtime/near-vm-runner/src/cache.rs [hunk 2]
```diff
@@ -111,12 +117,12 @@ impl fmt::Debug for MockCompiledContractCache {
 #[cfg(not(feature = "no_cache"))]
 const CACHE_SIZE: usize = 128;
```
Comment (matklad) [inline runtime/near-vm-runner/src/cache.rs:118]: Thias *might* give unused if both are disabled?

**Review by matklad — APPROVED** [review]
Heh, I think this is now significantly less painful to achieve than it used to be, thanks to your Runtime refactor!

One obvious question is "should we enforce this on CI"? I'd say at this point, as this requires an extra CI job, it probably isn't worth it. My bet is that, aftre wasmtime upgrade and wasmer0 removal, we should work on removing default features, which would make extra CI unnecessary.

File: runtime/near-vm-runner/src/cache.rs (hunk 2 above)
Comment (nagisa) [reply runtime/near-vm-runner/src/cache.rs:118]: There are a fair amount of unused lints when none of the implementations are enabled, including for this item. I'm not entirely sure it is worth the effort to go and clean up everything here, as I only really needed this for some hacking with aurora contracts.

I don't anticipate disabling all the runtimes will be all that useful thing to need.

**Pull request merged**